### DISCUSSION TITLE
Markdown vs Quarto (file type in VSCode)

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#1289] [Q&A]
I use VSCode extensions for markdown files, but I think they're not working in .qmd files because they're no longer Markdown files, but Quarto files (and there's an extension for it). For example, with [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) I can select text, and if my clipboard has a hyperlink, I can paste over that texte and the extension transforms the hyperlink properly in Markdown (saves a lot of clicks). 

Is there a general work-around to this? That is, Quarto should also be Markdown in terms of features in VSCode. 

I'm not a VSCode expert. Thanks for any insight.

**Comment by jjallaire** [comment]
The problem is that many of the existing markdown behaviors don't work as expected with Quarto (e.g. realtime preview has no idea how to deal with many of our markdown constructs). So we generally want extensions to "opt in" to working w/ Quarto once they've confirmed that things go smoothly for users. It's typically as simple as requesting (or providing a PR) that an extension add the "quarto" file type to the ones they work with.

**Reply by fuhrmanator** [reply]
OK, in the mean time (which I think will be a while), I can "force" the file format to "markdown" by clicking on the "Select language mode" at the lower right bar of the editor:
![image](https://user-images.githubusercontent.com/7606540/177363696-d0812643-2a62-4861-ab64-88a3f758eb55.png)

**Comment by jjallaire** [comment]
Yes, you can do that, but will lose pretty much all of the features of the Quarto extension at that point.
